var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -156,6 +156,20 @@ def configure_args(self):
 
         return ["--with-zlib=%s" % spec['zlib'].prefix] + extra_args
 
+    def configure(self, spec, prefix):
+        # Run the default autotools package configure
+        super(Hdf5, self).configure(spec, prefix)
+
+        if '@:1.8.14' in spec:
```

Comment:
I think people were arguing about arguments when we first started talking about the different build systems.  After using `pytest` a bit -- what if we made package-level functions injected?  e.g. I could write:

```python
    def install(self):
        pass
```

And that would be ok -- it just wouldn't be passed the spec or prefix, but I could also write:

```python
    def install(self, prefix, spec, stage, version):
        pass
```

And get the appropriate variables passed in based on their name (like the way fixtures are injected).  Thoughts?  Too fancy?  Not sure how familiar most packagers will be with arg injection but it's super powerful once you start using it.